port/win/env_default.cc

```diff
@@ -29,11 +26,12 @@ namespace {
   std::once_flag winenv_once_flag;
   Env* envptr;
 };
-
 }
 
 Env* Env::Default() {
   using namespace port;
+  ThreadLocalPtr::InitSingletons();
+  CompressionContextCache::InitSingleton();
```

Comment:
Oh interesting, we'll probably need this in PosixEnv too. And you'll probably need #3951 here.

Reply:
@ajkr noted. For Windows it is more for consistency. Currently the env is never destroyed but it does not experience the problem, providing all the tasks are timely shutdown.